src/app/details/details.component.html

```diff
@@ -1,7 +1,13 @@
-<h1>{{ userData.name }}</h1>
+<ng-container *ngIf="user$ | async as user; else error">
```

Comment:
There is a lot going on in this one line.

Because the component stores a reference to the observable ```user$```, we can use the angular ```async``` pipe to "unpack" the observable into the ```user``` variable which we can use throughout the rest of the template

If the observable has some error (HTTP connection error, etc), then we will show the ```errror``` template instead of the main one